src/test/bridges/yearn/YearnUnit.t.sol

```diff
@@ -0,0 +1,210 @@
+
+        // Some more labels
+        IYearnRegistry _registry = bridge.YEARN_REGISTRY();
+        address yvDAI = _registry.latestVault(address(DAI));
+        vm.label(address(_registry), "YEARN_REGISTRY");
+        vm.label(address(DAI), "DAI");
+        vm.label(address(WETH), "WETH");
+        vm.label(address(yvDAI), "yvDAI");
+        vm.label(address(OLD_YVETH), "old_yvETH");
+        vm.label(address(YVETH), "yvETH");
+    }
+
+    function testInvalidCaller(address _callerAddress) public {
+        vm.assume(_callerAddress != rollupProcessor);
+        // Use HEVM cheatcode to call from a different address than is address(this)
+        vm.prank(_callerAddress);
+        vm.expectRevert(ErrorLib.InvalidCaller.selector);
+        bridge.convert(emptyAsset, emptyAsset, emptyAsset, emptyAsset, 0, 0, 0, address(0));
+    }
+
+    function testInvalidInputAmount() public {
+        AztecTypes.AztecAsset memory ethAsset = AztecTypes.AztecAsset({
+            id: 1,
+            erc20Address: address(0x0000000000000000000000000000000000000000),
+            assetType: AztecTypes.AztecAssetType.ETH
+        });
+
+        vm.prank(rollupProcessor);
+        vm.expectRevert(ErrorLib.InvalidInputAmount.selector);
+        bridge.convert(ethAsset, emptyAsset, emptyAsset, emptyAsset, 0, 0, 1, address(0));
+    }
+
+    function testInvalidOutputAsset() public {
+        vm.prank(rollupProcessor);
+        vm.expectRevert(ErrorLib.InvalidOutputA.selector);
+        bridge.convert(emptyAsset, emptyAsset, emptyAsset, emptyAsset, 1, 0, 0, address(0));
+    }
+
+    function testInvalidNoValueETH() public {
+        AztecTypes.AztecAsset memory depositInputAssetA = AztecTypes.AztecAsset({
+            id: 0,
+            erc20Address: address(0),
+            assetType: AztecTypes.AztecAssetType.ETH
+        });
+        AztecTypes.AztecAsset memory depositOutputAssetA = AztecTypes.AztecAsset({
+            id: 100,
+            erc20Address: address(OLD_YVETH),
+            assetType: AztecTypes.AztecAssetType.ERC20
+        });
+
+        vm.prank(rollupProcessor);
+        vm.expectRevert(ErrorLib.InvalidInputAmount.selector);
+        bridge.convert(depositInputAssetA, emptyAsset, depositOutputAssetA, emptyAsset, 100, 0, 0, address(0));
+    }
+
+    function testIncorrectAuxData() public {
+        vm.prank(rollupProcessor);
+        vm.expectRevert(ErrorLib.InvalidAuxData.selector);
+        bridge.convert(emptyAsset, emptyAsset, emptyAsset, emptyAsset, 1, 0, 2, address(0));
+    }
+
+    // @dev In order to avoid overflows we set _depositAmount to be uint96 instead of uint256.
+    function testYearnBridgeUnitTestERC20(uint96 _depositAmount) public {
+        IYearnRegistry _registry = bridge.YEARN_REGISTRY();
+        address vault = _registry.latestVault(address(DAI));
+        uint256 availableDepositLimit = IYearnVault(vault).availableDepositLimit();
+        vm.assume(_depositAmount > 1 && _depositAmount < availableDepositLimit);
```

Comment:
You can use `bound` from the test contract instead. This might run into issues if the depositLimit is small as assume will fail a lot.